### PR TITLE
Event list

### PR DESCRIPTION
**Pull request by matthinz**
image

Work in progress on event listing page. See [Trello](https://trello.com/c/Tk6NVBEF/164-oite-event-creator-create-event-listing-for-specific-audiences)

## Things this does

- Attempts to make the "preamble" section on the events page editable in the CMS by looking up the corresponding `Page` model for `/events`
- Lists all events that _have not happened yet_, without pagination
- Uses some slightly janky, very MVP checkboxes for filtering, probably will want a consult with @dluetger and @michelle-rago about what filtering / searching options we'll want to prototype & test.

### CODE REVIEW

**Comment by matthinz** [comment]
@rahearn This is failing the ZAP scan because of lack of CSRF on the `<form>` I'm using for filtering. I don't think this form needs any anti-CSRF protection, but also a form may not end up being the mechanism for this filtering. But other than that I think this is ready for you to take a look.

**Review by rahearn — APPROVED** [review]
I ignored the CSRF finding for the events page to get CI passing.

None of the suggestions are blocking, though I think the `link_to` one would be a good idea.

**Comment by rahearn** [comment]
Forgot to add my other thoughts to the review.

1) I have some concerns about the checkbox UX, but I'm totally fine with it for now and adjust after research/design.
2) Overall, I really like the design you went with and think it's a solid foundation to build on.

File: _settings/audiences.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+audiences:
```
[the remark this answers is not in the document]
Comment (matthinz) [reply _settings/audiences.yml:1]: I think this kind of app data is good to put in .yml files, and it's relatively easy to enable editing if you need to.

File: app/javascript/application.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
 // Entry point for the build script in your package.json
 
 import "uswds";
+
+// Auto-submit forms on element clicks
+document.addEventListener("click", (evt) => {
```
[the remark this answers is not in the document]
Comment (matthinz) [reply app/javascript/application.js:6]: With vanilla js, if you're listening on a general selector, like here, it's typically more efficient to add a single event listener to the document than adding a listener to each element that matches the selector. (These events will typically bubble up to the document level.) This also covers the case where you want to catch clicks on things that don't exist in the DOM at the time the .js code is called.

**Comment by matthinz** [comment]
Yeah, I am 100% not intending for this to be the final UX.

**Pull request merged**